lisa/tests/scheduler/load_tracking.py

```diff
@@ -889,15 +889,15 @@ class CPUMigrationBase(LoadTrackingBase):
     """
 
     @classmethod
-    def _run_rtapp(cls, target, res_dir, profile, ftrace_coll):
+    def _run_rtapp(cls, target, res_dir, profile, ftrace_coll, cgroup=None):
```

Comment:
> Did not followed all the previous discussion thus get this with a pinch of salt and just let me know if we already agree on the following points:

My main point previously was regarding where the cgroup conf lives. It does not need to be passed to `TestBundle.__init__` (just like `rtapp_profile` which is now just a property computed out of `plat_info`, which is part of the "core state" of the object).

> Are we happy about adding yet another param this call?

Mostly yes, `_run_rtapp()` is a "standalone" class method that runs the thing and collects the stuff. It's quite easy to spot which class is making a direct use of `_run_rtapp()`, so it's easy to spot which class is "deviating" from the basic usage (in this case by using cgroups).

> How do we know in which order ftrace_coll or cgroups should be called? Are we forcing an ordering?

Ordering is the order in which the context managers are passed to `with` statement. We should probably only start tracing after cgroups are setup, since there is little value in tracing devlib's scripts.

> There is not the risk we keep adding new params to this call whenever we need to pass in something more?

Yes, as long as they are optional like this one it is not an issue. It's just that `_run_rtapp()` will offer more services if needed.

> Cannot we come up with a more generic approach, e.g. ordered list of context managers?

That could be an interesting structure, as long as the context manager abstraction is enough to describe the feature (i.e. it is fully self-contained). In the cgroup case, it is unfortunately not enough, so at least for this one we cannot put it in a CM black box:
```
as_root = cgroup is not None
```
FWIW the idea of an ordered list of CM is easy to implement using `contextlib.ExitStack`: https://docs.python.org/3/library/contextlib.html#contextlib.ExitStack